### PR TITLE
feat(message-hub): add astarte-message-hub package

### LINKED ISSUE #9: Add Astarte-message-hub package

### PR DESCRIPTION
**Pull request by joshuachp**
This will add the astarte-message-hub package to the buildroot.

Closes #9.

### CODE REVIEW

File: package/astarte-message-hub/astarte-message-hub.service [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Description=Astarte Message Hub
+After=syslog.target network-online.target remote-fs.target nss-lookup.target
+Wants=network-online.target
+# For development, keep restarting to wait for the platform to come online
+StartLimitIntervalSec=0
+
+[Service]
+Type=simple
+Environment="RUST_LOG=debug"
+ExecStart=/usr/bin/edgehog-device-runtime -c /etc/edgehog/edgehog-device-runtime-config.toml
```
Comment (harlem88) [inline package/astarte-message-hub/astarte-message-hub.service]:
```suggestion
ExecStart=/usr/bin/astarte-message-hub -c /etc/astarte/astarte-message-hub-config.toml
```

File: package/astarte-message-hub/astarte-message-hub.mk [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ASTARTE_MESSAGE_HUB_DEPENDENCIES = host-protobuf
+
+ASTARTE_MESSAGE_HUB_CARGO_ENV = \
+HOST_CC="x86_64-linux-gnu-gcc" \
+TARGET_CC="$(HOST_DIR)/bin/aarch64-linux-gcc" \
+TARGET_AR="$(HOST_DIR)/bin/aarch64-linux-ar"
+
+ASTARTE_MESSAGE_HUB_CARGO_BUILD_OPTS=
+
+ASTARTE_MESSAGE_HUB_CONFIG_TOML_FILE= $(TARGET_DIR)/etc/astarte-message-hub/config.toml
```
Comment (harlem88) [inline package/astarte-message-hub/astarte-message-hub.mk]:
```suggestion
ASTARTE_MESSAGE_HUB_CONFIG_TOML_FILE= $(TARGET_DIR)/etc/astarte/astarte-message-hub-config.toml
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: package/astarte-message-hub/astarte-message-hub.service (hunk 1 above)
Comment (joshuachp) [reply package/astarte-message-hub/astarte-message-hub.service]: Done

File: package/astarte-message-hub/astarte-message-hub.mk (hunk 2 above)
Comment (joshuachp) [reply package/astarte-message-hub/astarte-message-hub.mk]: Done

File: package/astarte-message-hub/Config.in [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+            Can be obtained with ./generate-astarte-credentials -t pairing -p yourrealmkey.key
+
+    config BR2_PACKAGE_ASTARTE_MESSAGE_HUB_ASTARTE_DEVICE_ID
+        string "Astarte Device ID"
+        default ""
+        help
+            Astarte identifies each device with a 128 bit Device ID which has to be unique within
+            its Realm
+
```
Comment (harlem88) [inline package/astarte-message-hub/Config.in:51]: Add also the option `grpc_socket_port` that is required for the astarte-message-hub. 
https://github.com/astarte-platform/astarte-message-hub/blob/b870447f882183574531d90fa558f8e52c8b153c/README.md?plain=1#L43

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: package/astarte-message-hub/Config.in (hunk 3 above)
Comment (joshuachp) [reply package/astarte-message-hub/Config.in:51]: Done

File: package/astarte-message-hub/Config.in [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   config BR2_PACKAGE_ASTARTE_MESSAGE_HUB_GRPC_PORT
+      string "Astarte Message Hub gRPC port"
+      default "50051"
```
Comment (harlem88) [inline package/astarte-message-hub/Config.in]: grpc_port is a `int value`.
```suggestion
      int "Astarte Message Hub gRPC port"
      default 50051
      range 1 65535
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: package/astarte-message-hub/Config.in (hunk 4 above)
Comment (joshuachp) [reply package/astarte-message-hub/Config.in]: Done

File: package/astarte-message-hub/astarte-message-hub.mk [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+###############################################################################
+#
+# ASTARTE_MESSAGE_HUB
+#
+################################################################################
+
+ASTARTE_MESSAGE_HUB_VERSION = master
```
Comment (harlem88) [inline package/astarte-message-hub/astarte-message-hub.mk]:
```suggestion
ASTARTE_MESSAGE_HUB_VERSION = v0.5.2
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: package/astarte-message-hub/astarte-message-hub.mk (hunk 5 above)
Comment (joshuachp) [reply package/astarte-message-hub/astarte-message-hub.mk]: Done

**Review by harlem88 — APPROVED** [review]

**Pull request merged**